litex/soc/software/liblitesdcard/sdcard.c

```diff
@@ -504,7 +504,7 @@ void sdcard_write(uint32_t block, uint32_t count, uint8_t* buf)
 	while (count--) {
 		/* Initialize DMA Reader */
 		sdmem2block_dma_enable_write(0);
-		sdmem2block_dma_base_write((uint64_t) buf);
+		sdmem2block_dma_base_write((uint64_t)(uintptr_t) buf);
```

Comment:
I'm building this for 64-bit Rocket, and I'm getting this warning (that I didn't use to get before):

```
 CC       sdcard.o
/home/somlo/LITEX/litex/litex/soc/software/liblitesdcard/sdcard.c: In function 'sdcard_read':
/home/somlo/LITEX/litex/litex/soc/software/liblitesdcard/sdcard.c:476:39: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  476 |  sdblock2mem_dma_base_write((uint64_t)(uintptr_t) buf);
      |                                       ^
/home/somlo/LITEX/litex/litex/soc/software/liblitesdcard/sdcard.c: In function 'sdcard_write':
/home/somlo/LITEX/litex/litex/soc/software/liblitesdcard/sdcard.c:507:40: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  507 |   sdmem2block_dma_base_write((uint64_t)(uintptr_t) buf);
      |                                        ^
```

Is there a way to silence the warnings on *both* 32- and 64-bit builds?

Reply:
The typedefs for (u)intptr_t in litex/soc/software/include/base/stdint.h are wrong:

> typedef int intptr_t;
> typedef unsigned int uintptr_t;

On a 64-bit system, they should use long, not int, cfr. /usr/include/stdint.h.